app/src/screens/UpdateWishlist.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect } from 'react';
+import Swal from 'sweetalert2';
+import { Dialog, DialogTitle, DialogPanel } from '@headlessui/react';
+import { useNavigate } from 'react-router-dom';
+import '../styles/CreateWishlist.css';
+
+function UpdateWishlist({ userKey, children, className }) {
+    const navigate = useNavigate();
+
+    const [wishlistName, setWishlistName] = useState('');
+    const [wishlistDescription, setWishlistDescription] = useState('');
+    const [loading, setLoading] = useState(true);
+    const [isOpen, setIsOpen] = useState(false); 
+
+    const fetchWishlist = async () => {
+        try {
+            const storedWishlistName = localStorage.getItem('wishlistName');
+            const storedWishlistDescription = localStorage.getItem('wishlistDescription');
+
+            if (storedWishlistName && storedWishlistDescription) {
+                setWishlistName(storedWishlistName);
+                setWishlistDescription(storedWishlistDescription);
+            } else {
+                const response = await fetch(`https://jomo-se-722e825d9259.herokuapp.com/api/wishlist/get-users/${userKey}`);
+                const wishlist = await response.json();
+
+                if (wishlist.length > 0) {
+                    setWishlistName(wishlist[0].name);
+                    setWishlistDescription(wishlist[0].description);
+                } else {
+                    Swal.fire({
+                        icon: 'error',
+                        title: 'No Wishlist Found',
+                        text: 'Please create a wishlist first',
+                    });
+                    navigate('/homepage');
+                }
+            }
+        } catch (error) {
+            console.error('Error fetching wishlist:', error);
+            Swal.fire({
+                icon: 'error',
+                title: 'Error',
+                text: 'An error occurred while fetching your wishlist.',
+            });
+        } finally {
+            setLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        fetchWishlist();
+    }, [userKey]);
+
+    const handleSave = async () => {
+        if (wishlistName.trim() === '' || wishlistDescription.trim() === '') {
+            Swal.fire({
+                icon: 'error',
+                title: 'Oops...',
+                text: 'Please enter both wishlist name and description.',
+            });
+            return;
+        }
+
+        const updatedWishlist = {
+            name: wishlistName,
+            description: wishlistDescription,
+            userKey: userKey,
+        };
+
+        try {
+            const response = await fetch(`https://jomo-se-722e825d9259.herokuapp.com/api/wishlist/update/${userKey}`, {
```

Comment:
Not sure this is right. This endpoint takes in a wishlist id to know which wishlist to update, not a user key. I could be wrong but You want to get which wishlist belongs to the user then update that wishlist.  We could talk about it a bit more but I can also create an endpoint that will specifically update a wishlist based off the userKey.  Let me know how you want to proceed.